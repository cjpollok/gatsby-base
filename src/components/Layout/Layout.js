import React from 'react'
import PropTypes from 'prop-types'
import { Helmet } from 'react-helmet'

import Header from '../Header'
import Footer from '../Footer'

import '../../styles/main.scss'
import styles from './layout.module.scss'
import favicon from '../../images/favicon.ico'

const Layout = ({ children }) => (
  <div className={styles.container}>
    <Helmet defaultTitle="Gatsby Base" titleTemplate="%s | Gatsby Base">
      <html lang="en" />
      <link rel="icon" type="image/x-icon" href={favicon} />
      <meta
        name="description"
        content="A base Gatsby setup for use in other projects"
      />
    </Helmet>
    <Header siteTitle="Gatsby Base" />
    {children}
    <Footer />
  </div>
)

Layout.propTypes = {
  children: PropTypes.node.isRequired,
}

export default Layout
