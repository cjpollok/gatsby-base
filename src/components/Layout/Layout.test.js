import React from 'react'
import { shallow } from 'enzyme'

import Layout from './Layout'

describe('Layout Component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Layout>
        <p>Test Child</p>
      </Layout>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
