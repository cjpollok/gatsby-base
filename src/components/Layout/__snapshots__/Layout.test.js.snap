// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component renders correctly 1`] = `
<div
  className="container"
>
  <HelmetWrapper
    defaultTitle="Gatsby Base"
    defer={true}
    encodeSpecialCharacters={true}
    titleTemplate="%s | Gatsby Base"
  >
    <html
      lang="en"
    />
    <link
      href="test-file-stub"
      rel="icon"
      type="image/x-icon"
    />
    <meta
      content="A base Gatsby setup for use in other projects"
      name="description"
    />
  </HelmetWrapper>
  <Header
    siteTitle="Gatsby Base"
  />
  <p>
    Test Child
  </p>
  <Footer />
</div>
`;
